devito/types/basic.py

```diff
@@ -463,6 +463,61 @@ class AbstractFunction(sympy.Function, Basic, Pickable):
     is_AbstractFunction = True
 
 
+class AbstraceCachedAnonymousFunction(AbstractFunction, Cached, Evaluable):
```

Comment:
why can you not just use AbstractCachedFunction, from which Array ( a data-free tensor symbol) inherits?

Reply:
There is a lot of stuff (like halo/shape/indices) setups that are unnecessary in AbstractCachedFunction that make it nasty.

Reply:
override and make them no-op?

Reply:
I think we should avoid a new Abstract class at all costs (btw, typo: Abstrace...)